src/applications/common/schemaform/save-in-progress/helpers.js

```diff
@@ -0,0 +1,63 @@
+import {
+  createPageList,
+  createFormPageList,
+  createRoutes
+} from '../helpers';
+import RoutedSavablePage from './RoutedSavablePage';
+import RoutedSavableReviewPage from './RoutedSavableReviewPage';
+import FormSaved from './FormSaved';
+import SaveInProgressErrorPage from './SaveInProgressErrorPage';
+
+export function addSaveInProgressRoutes(config) {
+  const protectedRoutes = new Set(['introduction', 'review-and-submit', 'confirmation', '*']);
+  const formPages = createFormPageList(config);
+  const pageList = createPageList(config, formPages);
+  const newRoutes = createRoutes(config);
+
+  newRoutes.forEach((route, index) => {
+    let newRoute;
+
+    // rewrite page component
+    if (!protectedRoutes.has(route.path)) {
+      newRoute = Object.assign({}, route, { component: RoutedSavablePage });
+      if (index !== -1) {
```

Comment:
When is the index set to `-1`?

Reply:
I can't remember why I added this... and looking at it it doesn't make sense. Going to just remove it.